### DISCUSSION TITLE
Dumb cameras vs. dumb NVRs

### DISCUSSION THREAD

**Discussion by thinkloop** [blakeblackshear/frigate#4830] [General]
One of the (many) nice things about Frigate is that you can throw up any crappy camera you find and get advanced, consistent features for things like object detection. However I just purchased a camera with auto-tracking where a motor in the camera moves to follow subjects throughout the frame. This requires that the camera have its own built-in intelligence - much of what Frigate does. As chipsets get cheaper and AI gets "solved" and features like auto-tracking become more in demand, will smart NVRs like Frigate inevitably start to be replaced by smart cameras and the NVR only focused on simple centralized recording? What are your thoughts on how this will play out?

**Comment by NickM-27** [comment]
Even the smart cameras today offer maybe a few different detections (person, car, animal) and it is very much "is there an object in this space". You can't even set specific zones so you could for example only turn the lights on when someone steps in your front porch even though you want to be notified when they step in your yard.

Also, that's not mentioning the many more labels that are possible with a solution like Frigate+ being able to detect Amazon, Fedex, UPS, etc.

Just like anything I think different consumers will have different priorities and different solutions will suite different sets of priorities.

**Comment by blakeblackshear** [comment]
I think standalone NVRs have the benefit of sharing compute resources across many cameras rather than needing to duplicate AI enabled hardware in each camera. They can also consider information from multiple cameras at once like moving a PTZ to an area where another camera detects an object.

I expect cameras to get more AI built in directly, but I think the sweet spot will be for limited use cases. PTZ tracking is an obvious one that benefits from AI on the device.

We will likely see a mix of these two in ways that complement each other. I don't really expect to see any of these options offer open APIs and MQTT to make it integrated into home assistant or other home automation platforms. Off the shelf systems will likely end up being good enough for most people's needs and Frigate will be an option for those who want deeper integration with HA or some of the more advanced features.

**Comment by hagak** [comment]
I am also of the thought that PTZ is used inapproiately in most instances.  Better to have more cameras covering and area.  For instance if your PTZ camera is covering an area but an object moves outside that area it will never see or track that object.  If the PTZ starts tracking an object but the real threat starts outside of frame what happens?  Only time I have seen PTZ implemented in a good way is having a wide angle camera covering an area and have it control a PTZ camera that has a longer focal length for things like "small object detection" and "license plate" reading.  But in this instance you could potentially make Frigate work if you have a PTZ camera with responsive controls (most are horribly slow).

**Comment by spacebares** [comment]
Reolink is certainly leaning in that direction, their NVRs are seriously lacking in features and offer no additional features to their cameras other then storage. 

Everything from thumbnail notifications _(newer cams)_ to AI object detection and storage _(SD card)_ can be done without their NVR, Viewing all cameras at once like an NVR could do can be done natively in the Reolink desktop APP using the split feature. 
Does it look like Reolink is moving away from centralized NVRs? Certainly looks that way...
Meanwhile Eufy is doubling down on NVR's with their Homebase3, offering new AI detection & facial recognition threw its "NVR" for their new line of cameras.

Can frigate be replaced in the future by more modern systems? I can't speak for other camera systems, but certainly not by Reolink any time soon. Those who have a Reolink system knows just how bad its lacking in usability.
Frigate offers a ton of customization. The main thing that brought me to frigate was object detection via different zones and the event browser design.

Alongside Home Assistant, i can configure frigate to record all events, but only buzz phones if a person or car is detected in the driveway, or the Door zones, etc. On my phone i can view event thumbnails in the notifications. Reviewing some 20 events just by looking at thumbnails in under a minute without ever having to view footage is something you never thought you needed in a CCTV system until you see it for yourself. These events are all in important zones and not just the neighbor across the street. Its the mail man working the mailbox. Its amazon dropping a package off. Its the dogs being let out. I look at the thumbnail and know exactly what has happened without needing to look at footage.
And when i do see something interesting? Just click on it, now your watching a video of someone who is simply just trying to grab their dog who ran into your yard. Its all just so quick and easy to use. 

So now instead of having to return all these Reolink cameras and its NVR, they are now apart of Frigate. Best of all I'm not limited to one camera brand. If its got RTMP/RTSP frigate can use it. Along side my Reolink cameras I have two Eufy battery cameras also hooked up to frigate. Tell me what manufacturer lets you do this with their system?

If there is a problem with a name brand's system you have to open a ticket and hope that your problem gets pushed up some ladder and shows up in an update a couple months later. Since frigate is open source anyone can contribute changes and features. Anyone can help fix bugs or offer support. I wanted a way to sort favorite clips in the event browser so i [added it](https://github.com/blakeblackshear/frigate/pull/4737). Frigate was so well built, it only needed just 4 more small blocks of code. Its the little things like this that just keeps making frigate better as time goes on, and the whole of the internet can come in and help.

There is always going to be a place for Frigate because by design, Frigate is from the people, for the people. An open source NVR solution.
